accounts-db/src/accounts_db.rs

```diff
@@ -3409,35 +3461,45 @@ impl AccountsDb {
                 }
                 true
             });
+            }
         }
         store_counts_time.stop();
 
         let mut calc_deps_time = Measure::start("calc_deps");
-        Self::calc_delete_dependencies(&purges_zero_lamports, &mut store_counts, min_dirty_slot);
+        self.calc_delete_dependencies(&candidates, &mut store_counts, min_dirty_slot);
         calc_deps_time.stop();
 
         let mut purge_filter = Measure::start("purge_filter");
         self.filter_zero_lamport_clean_for_incremental_snapshots(
```

Comment:
we have to filter inside this fn as well, here:
https://github.com/anza-xyz/agave/blob/2b7efbf00bfecb82f205d24765d50926a2240a9c/accounts-db/src/accounts_db.rs#L3762

similar to the `filter_map(...)` added in this pr already recently.

Reply:
I added a check for empty slot_list. Will this work?

Reply:
yes